keepsorted/keep_sorted_test.go

```diff
@@ -973,6 +973,29 @@ func TestLineGrouping(t *testing.T) {
 				}},
 			},
 		},
+		{
+			name: "Group_HangingNewline",
```

Comment:
It took me longer than I'd like to admit to understand just what we were doing (and what a hanging newline meant). I am not sure if "hanging newline" is necessarily the best thing to call this. I don't have any snappy suggestion, but perhaps that indicates that we need to have a comment to elaborate on what we're testing.

IIUC, we're allowing _blank_ lines to be considered part of the group if the next non-blank line would be part of the group?

Reply:
I rephrased the commit description as well